lib/utils/update-notifier.js

```diff
@@ -113,3 +115,11 @@ const updateNotifier = module.exports = async (npm, spec = 'latest') => {
 
   return messagec
 }
+
+// only update the notification timeout if we actually finished checking
+module.exports = async npm => {
+  const notification = await updateNotifier(npm)
+  // intentional.  do not await this.  it's a best-effort update.
+  updateTimeout(npm)
+  return notification
+}
```

Comment:
```suggestion
module.exports = npm => {
  // intentional.  do not await this.  it's a best-effort update.
  updateTimeout(npm)
  return updateNotifier(npm)
}
```

Reply:
No, because we don't want to even start updating the timeout file until we know that the update check actually completed.

Reply:
Maybe less clever-looking approach would be:

```js
module.exports = npm => {
  return updateNotifier(npm).then(notification => {
    updateTimeout(npm)
    return notification
  })
}
```